src/app/shared/components/button/button.component.ts

```diff
@@ -12,6 +12,7 @@ import { fas } from '@fortawesome/free-solid-svg-icons';
 export class ButtonComponent implements OnChanges{
   @Input() buttonText?:string;
   @Input() iconName?:IconNames;
+  @Input() type?:'submit'|'button'= 'button';
```

Comment:
This calls for another enumeration I think. Across the templates where `<app-button>` is used, you should use it instead of the magic strings.
Also, make suer that all `<app-button>` tags include this `type` input.

Reply:
it is Done Thank you!
just 2 quick questions: 
1. When I use an Enum value I always need to import the Type and define as well in a Component to be able to use in it? Or is there a shorter way to see global enum types in all html?
2.  For the comment: Also, make suer that all <app-button> tags include this type input. -  I ahve done this but is it really necessary to give always the input? Isnt it a good practice to define a default value for the input and just dont pass anything when we want to use the default value?

Reply:
1. Yes, as far as I know there is not option to declare and export an Enum, Type or Interface somewhere and 'import and use' is directly in the templates. The control-model layer (component .ts) is responsible and serve as a bridge here. So import the needed enum, assign it to a variable and use it on the template.
2. Well, it depends on the use-case. I acknowledge that the button type is not subject to change. Generally speaking the input property is more likely to change.